e2e/cypress/integration/enterprise/system_console/openid/openid_spec.js

```diff
@@ -0,0 +1,244 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Group: @enterprise @system_console
+
+import * as TIMEOUTS from '../../../../fixtures/timeouts';
+import {hexToRgbArray, rgbArrayToString} from '../../../../utils';
+
+const FAKE_SETTING = '********************************';
+const SERVICE_PROIVDER_LABEL = 'Select Service Provider:';
+const DISCOVERY_ENDPOINT_LABEL = 'Discovery Endpoint:';
+const CLIENT_ID_LABEL = 'Client ID:';
+const CLIENT_SECRET_LABEL = 'Client Secret:';
+
+// # Goes to the System Scheme page as System Admin
+const goToAdminConsole = () => {
+    cy.apiAdminLogin();
+    cy.visit('/admin_console');
+};
+
+const verifyOAuthLogin = (id, text, color, href) => {
+    cy.apiLogout();
+
+    // cy.findByLabelText(`${text}`).then((btn) =>{
+    //     expect(btn[0].href).equal(href);
+    //     if (color) {
+    //         const rbgArr = hexToRgbArray(color);
+    //         expect(btn[0].style.backgroundColor).equal(rgbArrayToString(rbgArr));
+    //     }
+    // });
+    cy.get(`#${id}`).then((btn) => {
+        expect(btn[0].innerText).equal(`${text}`);
+        expect(btn[0].href).equal(href);
+        if (color) {
+            const rbgArr = hexToRgbArray(color);
+            expect(btn[0].style.backgroundColor).equal(rgbArrayToString(rbgArr));
+        }
+    });
+};
+
+describe('MM-27688 - System console-OpenId Connect', () => {
+    before(() => {
+        // * Check if server has license
+        cy.apiRequireLicense();
+        cy.apiUpdateConfig({
+            FeatureFlags: {
+                OpenIdConnect: 'on',
+            },
+        });
+    });
+    it('MM-27688 - Set to Generic OpenId', () => {
+        // # Go to admin console and set permissions as listed in the test
+        goToAdminConsole();
+
+        cy.findByTestId('authentication.openid').click();
```

Comment:
According to Test-Library best practise https://testing-library.com/docs/guide-which-query/
`cy.findByRole('link', { name: /openid connect/i }).click();`